### PR TITLE
Tori.teng/log search bar docs update

### PR DESCRIPTION
**Pull request by toriteng**
### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/logs/explorer/search.md [hunk 1]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+Use the search bar's autocomplete feature to complete your query using:
+- Existing keys and values in your logs
+- Your recent searches (recent searches from other users are not displayed)
+- Saved views
+
+{{< img src="logs/explorer/search/log_search_bar_autocomplete.png" alt="The logs search bar showing service: as the query and emailer, balancer-checker, ad-server, and vpc as autocomplete options" style="width:80%;">}}
+
+### Autocomplete facets and values
+
+The search bar autosuggests facets based on your input in the search bar. These facets are displayed in the same order in which they are positioned in the [facet panel][7]. If a facet has a defined dispay name, that is displayed on the righthand side of the dropdown. Facets that are not configured to be displayed in the facet panel are also not autosuggested during search.
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:46]:
```suggestion
The search bar autosuggests facets based on your input in the search bar. These facets are displayed in the same order in which they are positioned in the [facet panel][5]. If a facet has a defined display name, it is displayed on the right-hand side of the dropdown menu. Facets that are not configured to be displayed in the facet panel are not autosuggested for a search.
```

File: content/en/logs/explorer/search.md [hunk 2]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
+{{< img src="logs/explorer/search/log_facet_autocomplete.png" alt="The logs search bar showing `network` as the query and the facets @network.bytes_written, @network.client.ip, and @network.interface as autocomplete options" style="width:80%;">}}
+
+After you select a facet and input the `:` character, the search bar autosuggests values. These values are displayed in descending order of how many logs contain that facet:value pair in the past 15 minutes. The estimated number of logs containing that value is displayed on the righthand side of the dropdown. For example, the `balance-checker` service is positioned first in the autosuggested list of values for the `service` facet, indicated by 2.66M, the highest log count:
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:50]:
```suggestion
After you select a facet and input the `:` character, the search bar autosuggests values. These values are displayed in descending order of how many logs contain that `facet:value` pair in the past 15 minutes. The estimated number of logs containing that value is displayed on the right-hand side of the dropdown menu. For example, the `balance-checker` service is positioned first in the autosuggested list of values for the `service` facet, indicated by the 2.66M logs, the highest log count:
```

File: content/en/logs/explorer/search.md [hunk 3]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
+{{< img src="logs/explorer/search/log_value_autocomplete.png" alt="The logs search bar showing `service:` as the query and the values balance-checker, ad-server, fraud-detector, and trade-executor as autocomplete options" style="width:80%;">}}
+
+### Autocomplete recent searches
+
+Your 100 most recent searches run in the Log Explorer are retained. Recent searches from other users are not retained or displayed. The search bar autosuggests the 4 most recent searches that match your input in the search bar, with the most recent search displayed first. The search bar also shows how long ago each recent search was run. For example, if you input `service:web-store status:error` in the search bar, the 4 most recent searches containing these terms are displayed in order of recency, each one specifying a different error:
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:56]:
```suggestion
Your 100 most recent searches in the Log Explorer are retained. Recent searches from other users are not retained or displayed. The search bar autosuggests the four most recent searches that match your input in the search bar, with the most recent search displayed first. The search bar also shows how long ago each recent search was run. For example, if you input `service:web-store status:error` in the search bar, the four most recent searches containing these terms are displayed in order of recency, each one specifying a different error:
```

File: content/en/logs/explorer/search.md [hunk 4]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
+{{< img src="logs/explorer/search/log_recent_searches.png" alt="The logs search bar showing `service:web-store status:error` as the query and recent searches for different web-store service errors as autocomplete options" style="width:80%;">}}
+
+### Autocomplete Saved Views
+
+You can create Saved Views in the Log Explorer to save queries and additional context for future and centralized access. The search bar autosuggests Saved Views that match your input in the search bar. Saved Views are displayed in the same order in which they are positioned in the Saved Views panel, with Saved Views that have been favorited displayed first. The Saved View name, saved query, and profile picture of the user who last updated it are displayed in the dropdown. If a Saved View query is too long to be displayed in the dropdown, the full query is displayed in a tooltip on hover. The email of the user who last updated a Saved View is also displayed in a tooltip on hover over their profile picture.
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:62]:
```suggestion
You can create Saved Views in the Log Explorer to save queries and additional context for the future and for centralized access. The search bar autosuggests Saved Views that match your input in the search bar. Saved Views are displayed in the same order in which they are positioned in the Saved Views panel, with starred Saved Views displayed first. The Saved View name, saved query, and profile picture of the user who last updated it are displayed in the dropdown menu. If a Saved View query is too long to be displayed in the dropdown, the full query is displayed in a tooltip on hover. The email of the user who last updated a Saved View is also displayed in a tooltip on hover over their profile picture.
```

File: content/en/logs/explorer/search.md [hunk 5]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
+{{< img src="logs/explorer/search/log_autocomplete_saved_views.png" alt="The logs search bar showing `service:web-store status:error` as the query and Saved Views for different web-store service errors as autocomplete options" style="width:80%;">}}
+
 ## Search syntax
 
+Syntax highlighting clearly differentiates input types, such as keys (e.g. an attribute such as `@merchant_name`), values (e.g. the name of a particular merchant), free text (e.g. keywords in a log messages such as `responded 500`), and control characters (e.g. parantheses and colons). Status attributes are also highlighted in colors representing the status, such as red for `error` and blue for `info`.
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:68]:
```suggestion
Syntax highlighting clearly differentiates input types, such as keys (for example, an attribute such as `@merchant_name`), values (for example, the name of a particular merchant), free text (for example, keywords in a log messages such as `responded 500`), and control characters (for example, parentheses and colons). Status attributes are also highlighted in colors representing the status, such as red for `error` and blue for `info`.
```

File: content/en/logs/explorer/search.md [hunk 6]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
+{{< img src="logs/explorer/search/log_syntax_highlighting.png" alt="The logs search bar showing `service:auth-dotnet status:error 500 (check-token OR create-user)` as the query with differentiable syntax highlighting" style="width:80%;">}}
+
+Clear error states inform you which part of the query contains syntax errors and how to remediate them. For example,
+- If you input the query `service:` with no value, the message "Missing value in key:value pair" is displayed when you hover over the query.
+- If you input brackets for a range query, but do not fill in the high and low values, the message "Expected term but end of input found" is displayed.
+- If you input multiple values for a log field but miss the closing parenthesis character, such as  `service:(web-store OR auth-dotnet`, the message `Missing closing parenthesis character` is displayed:
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:75]:
```suggestion
- If you input multiple values for a log field but miss the closing parenthesis character, such as  `service:(web-store OR auth-dotnet`, the message `Missing closing parenthesis character` is displayed.
```

File: content/en/logs/explorer/search.md [hunk 7]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
 To start searching for logs and customizing the time frame in the Log Explorer, read the [Search Syntax documentation][3] and the [Custom Time Frames documentation][4].
 
+## Raw and user-friendly search mode
+
+The user-friendly search mode contains autocomplete and syntax highlighting as mentioned. Toggle the button to the right of the search bar to search in raw mode, where syntax highlighting, search pills styling, and autocomplete are removed:
+
+{{< img src="logs/explorer/search/log_raw_search_mode.png" alt="The logs search bar showing `service:auth-dotnet status:error 500 (check-token OR create-user)` as the query in raw search mode" style="width:80%;">}}
+
+You can interact with the search bar with your mouse, as well as by using keyboard interactions such as:
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:87]:
```suggestion
You can interact with the search bar with your mouse, as well as by using keyboard commands such as:
```

File: content/en/logs/explorer/search.md [hunk 8]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
+{{< img src="logs/explorer/search/log_autocomplete_saved_views.png" alt="The logs search bar showing `service:web-store status:error` as the query and Saved Views for different web-store service errors as autocomplete options" style="width:80%;">}}
+
 ## Search syntax
 
+Syntax highlighting clearly differentiates input types, such as keys (e.g. an attribute such as `@merchant_name`), values (e.g. the name of a particular merchant), free text (e.g. keywords in a log messages such as `responded 500`), and control characters (e.g. parantheses and colons). Status attributes are also highlighted in colors representing the status, such as red for `error` and blue for `info`.
+
+{{< img src="logs/explorer/search/log_syntax_highlighting.png" alt="The logs search bar showing `service:auth-dotnet status:error 500 (check-token OR create-user)` as the query with differentiable syntax highlighting" style="width:80%;">}}
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:70]:
```suggestion
{{< img src="logs/explorer/search/log_syntax_highlighting.png" alt="The logs search bar showing `service:auth-dotnet status:error 500 (check-token OR create-user)` as the query with differentiable syntax highlighting" style="width:100%;">}}
```

File: content/en/logs/explorer/search.md [hunk 9]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
+Clear error states inform you which part of the query contains syntax errors and how to remediate them. For example,
+- If you input the query `service:` with no value, the message "Missing value in key:value pair" is displayed when you hover over the query.
+- If you input brackets for a range query, but do not fill in the high and low values, the message "Expected term but end of input found" is displayed.
+- If you input multiple values for a log field but miss the closing parenthesis character, such as  `service:(web-store OR auth-dotnet`, the message `Missing closing parenthesis character` is displayed:
+
+{{< img src="logs/explorer/search/log_error_states.png" alt="The logs search bar showing `service:(web-store OR auth-dotnet` as the query with the message `Missing closing parenthesis character`" style="width:80%;">}}
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:77]:
```suggestion
{{< img src="logs/explorer/search/log_error_states.png" alt="The logs search bar showing `service:(web-store OR auth-dotnet` as the query with the message `Missing closing parenthesis character`" style="width:50%;">}}
```

File: content/en/logs/explorer/search.md [hunk 10]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
 To start searching for logs and customizing the time frame in the Log Explorer, read the [Search Syntax documentation][3] and the [Custom Time Frames documentation][4].
 
+## Raw and user-friendly search mode
+
+The user-friendly search mode contains autocomplete and syntax highlighting as mentioned. Toggle the button to the right of the search bar to search in raw mode, where syntax highlighting, search pills styling, and autocomplete are removed:
+
+{{< img src="logs/explorer/search/log_raw_search_mode.png" alt="The logs search bar showing `service:auth-dotnet status:error 500 (check-token OR create-user)` as the query in raw search mode" style="width:80%;">}}
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:85]:
```suggestion
{{< img src="logs/explorer/search/log_raw_search_mode.png" alt="The logs search bar showing `service:auth-dotnet status:error 500 (check-token OR create-user)` as the query in raw search mode" style="width:100%;">}}
```

**Review by maycmlee — COMMENTED** [review]
Left some suggestions!

File: content/en/logs/explorer/search.md [hunk 11]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+Clear error states inform you which part of the query contains syntax errors and how to remediate them. For example,
+- If you input the query `service:` with no value, the message "Missing value in key:value pair" is displayed when you hover over the query.
+- If you input brackets for a range query, but do not fill in the high and low values, the message "Expected term but end of input found" is displayed.
+- If you input multiple values for a log field but miss the closing parenthesis character, such as  `service:(web-store OR auth-dotnet`, the message `Missing closing parenthesis character` is displayed:
+
+{{< img src="logs/explorer/search/log_error_states.png" alt="The logs search bar showing `service:(web-store OR auth-dotnet` as the query with the message `Missing closing parenthesis character`" style="width:80%;">}}
+
 To start searching for logs and customizing the time frame in the Log Explorer, read the [Search Syntax documentation][3] and the [Custom Time Frames documentation][4].
 
+## Raw and user-friendly search mode
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:81]: Pranay said to change this to:
```suggestion
## Disable styling for search bar
```

File: content/en/logs/explorer/search.md [hunk 12]
```diff
@@ -32,10 +32,64 @@ For example, to filter on logs produced by a web store service, with an error st
...
+
+The user-friendly search mode contains autocomplete and syntax highlighting as mentioned. Toggle the button to the right of the search bar to search in raw mode, where syntax highlighting, search pills styling, and autocomplete are removed:
+
+{{< img src="logs/explorer/search/log_raw_search_mode.png" alt="The logs search bar showing `service:auth-dotnet status:error 500 (check-token OR create-user)` as the query in raw search mode" style="width:80%;">}}
+
+You can interact with the search bar with your mouse, as well as by using keyboard interactions such as:
+- CMD-A for selecting
+- CMD-C for copying
+- CMD-X for cutting
+- CMD-V for pasting
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:91]:
```suggestion
```

File: content/en/logs/explorer/search.md (hunk 11 above)
Comment (toriteng) [reply content/en/logs/explorer/search.md:81]: I think we should mention that users can switch to raw text mode explicitly.

File: content/en/logs/explorer/search.md [hunk 13]
```diff
@@ -32,10 +32,60 @@ For example, to filter on logs produced by a web store service, with an error st
...
+- If you input brackets for a range query, but do not fill in the high and low values, the message "Expected term but end of input found" is displayed.
+- If you input multiple values for a log field but miss the closing parenthesis character, such as  `service:(web-store OR auth-dotnet`, the message `Missing closing parenthesis character` is displayed.
+
+{{< img src="logs/explorer/search/log_error_states.png" alt="The logs search bar showing `service:(web-store OR auth-dotnet` as the query with the message `Missing closing parenthesis character`" style="width:50%;">}}
+
 To start searching for logs and customizing the time frame in the Log Explorer, read the [Search Syntax documentation][3] and the [Custom Time Frames documentation][4].
 
+## Raw and user-friendly search mode
+
+The user-friendly search mode contains autocomplete and syntax highlighting as mentioned. Toggle the button to the right of the search bar to search in raw mode, where syntax highlighting, search pills styling, and autocomplete are removed:
```
Comment (maycmlee) [inline content/en/logs/explorer/search.md:83]:
```suggestion
Toggle the button to the right of the search bar to search in raw mode, where syntax highlighting, search pills styling, and autocomplete are removed:
```

**Review by maycmlee — COMMENTED** [review]
Also, one edit to take out this sentence because I think we can just jump directly to how to toggle the raw search mode.

File: content/en/logs/explorer/search.md (hunk 11 above)
Comment (maycmlee) [reply content/en/logs/explorer/search.md]: Updated to what was decided on

**Review by maycmlee — APPROVED** [review]

**Pull request merged**